src/gpuarray_elemwise.c

```diff
@@ -31,6 +31,7 @@ struct _GpuElemwise {
 STATIC_ASSERT(GEN_CONVERT_F16 == GE_CONVERT_F16, same_flags_value_elem1);
 
 #define is_array(a) (ISCLR((a).flags, GE_SCALAR))
+#define is_rw_array(a) (ISSET((a).flags, GE_READ) && ISSET((a).flags, GE_WRITE))
```

Comment:
Currently, the 1st dimension of matrix_a will be "repeated" 5 times: for example:

```
[[1,2,3,4]]
```
becomes
```
[[1,2,3,4]
[1,2,3,4]
[1,2,3,4]
[1,2,3,4]
[1,2,3,4]]
```
And then this temp matrix is added to matrix_b and then the result is stored in matrix_c (so matrix_c will be a matrix of shape (5,4)). That's how NumPy does it currently on my computer! (and pygpu also, for the moment).

So, do we disallow that behavior for GpuElemwise ?